### PR TITLE
Updating java.file.attribute

### PR DESCRIPTION
**Pull request by albertomisail**

### CODE REVIEW

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/nio/file/attribute/UserPrincipalNotFoundExceptionTest.java [hunk 1]
```diff
@@ -17,13 +17,11 @@
...
 
 import org.junit.Test;
-import junit.framework.TestCase;
-
 import static org.junit.Assert.assertEquals;
 
 import java.nio.file.attribute.UserPrincipalNotFoundException;
 
-public class UserPrincipalNotFoundExceptionTest extends TestCase {
+public class UserPrincipalNotFoundExceptionTest {
```
Comment (tomball) [inline jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/nio/file/attribute/UserPrincipalNotFoundExceptionTest.java:24]: Why the change? Were JUnit4 tests not being run because the runner categorized this as a JUnit3 test class?

File: jre_emul/Classes/java/nio/file/attribute/FileTime.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
  *  limitations under the License.
  */
 
-package com.google.j2objc.nio.file.attribute;
+package java.nio.file.attribute;
 
 import java.util.concurrent.TimeUnit;
```
Comment (tomball) [inline jre_emul/Classes/java/nio/file/attribute/FileTime.java:21]: Please work with Mary to expand the class doc-comment below (GitHub review UI won't let me comment there), to fully explain that this is a complete j2objc rewrite, designed to avoid Android's FileTime java.time dependencies.

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/nio/file/attribute/UserPrincipalNotFoundExceptionTest.java (hunk 1 above)
Comment (albertomisail) [reply jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/nio/file/attribute/UserPrincipalNotFoundExceptionTest.java:24]: The previous version was added by mistake in libcore.java.nio.channels.attribute. Had to delete it and add it back in the correct place. The old file in the incorrect location was not being found by JUnit unless `extends TestCase` was added. Now that the file is in the correct folder there is no need for the extend and file can be identical to libcore version

File: jre_emul/Classes/java/nio/file/attribute/FileTime.java (hunk 2 above)
Comment (albertomisail) [reply jre_emul/Classes/java/nio/file/attribute/FileTime.java:21]: Done

File: jre_emul/Classes/java/nio/file/attribute/FileTime.java [hunk 3]
```diff
@@ -22,6 +22,10 @@
 /**
  * Convert FileTime units.
  *
+ * This is a complete j2objc rewrite, designed to avoid Android's
+ * FileTime java.time dependencies. If an app only needs jre_channels (which many do),
+ * adding java.nio.file.FileTime to jre_channels would increase its size by ~23M.
+ *
```
Comment (tomball) [inline jre_emul/Classes/java/nio/file/attribute/FileTime.java:28]: Perfect, thanks.
